### PR TITLE
Improve spawn placement routines to be bounded in time to a single frame

### PR DESCRIPTION
**Pull request by dagmorland**

### CODE REVIEW

File: src/bank00.asm [hunk 1]
```diff
@@ -7087,23 +7099,147 @@ ELIF DEF(RNG_LCG)
...
-db $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00,
-db $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00,
-db $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00,
-db $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00,
-db $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00,
-db $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00,
-db $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00,
-db $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00,
-db $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00, $00
+; Get a uniform random number on [0,C)
```
Comment (xenophile127) [inline src/bank00.asm]: Please move this to $1c44 and leave the RNG area untouched.

File: src/bank03_npc.asm [hunk 2]
```diff
@@ -821,6 +821,7 @@ setNpcSpawnTable:
     pop  HL                                            ;; 03:4486 $e1
     ret                                                ;; 03:4487 $c9
 
+IF DEF(RNG_ORIGINAL)
```
Comment (xenophile127) [inline src/bank03_npc.asm]: With the bank 00 changes moved to $1c44 this conditional should not be necessary, correct?

File: src/memory.asm [hunk 3]
```diff
@@ -1589,12 +1592,21 @@ wD8DA:
...
+; First byte holds number of options (up to 165)
+; Subsequent 2-byte pairs provide a spawn location option (x followed by y).
+; This space can be reused for scratch space in its entirety outside of the spawnNpcsFromTable function.
+wSpawnPlacementScratch:
+    ds 331
 
 ; Free WRAM space.
 wFree:
-    ds 1825                                            ;; d8df
+    ds 1335                                            ;; d8df
```
Comment (xenophile127) [inline src/memory.asm]: Please correct or remove the positional comment.

Comment (dagmorland) [reply src/memory.asm]: Fixed

File: src/bank03_npc.asm (hunk 2 above)
Comment (dagmorland) [reply src/bank03_npc.asm]: Correct, I created a new SPAWN define in the Makefile to switch back and forth instead. We can remove when we're ready.

File: src/bank00.asm (hunk 1 above)
Comment (dagmorland) [reply src/bank00.asm]: Moved

**Pull request merged**